.github/workflows/main.yaml

```diff
@@ -72,13 +76,9 @@ jobs:
 
   publish-charm:
     name: Publish Charm
-    needs:
-      - lint-report
-      - static-analysis
-      - unit-tests-with-coverage
-      - integration-test
+    needs: integration-test
     runs-on: ubuntu-22.04
-    if: ${{ github.ref_name == 'main' }}
```

Comment:
I think we need to push the charm only for main branch.

Reply:
That's already covered with the top-level:
```yaml
on:
  push:
    branches: ["main"]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: ["main"]
```